### PR TITLE
PHOENIX-1981 : PhoenixHBase Load and Store Funcs should handle all Pig data types

### PR DESCRIPTION
**Pull request by prashantkommireddi**

### CODE REVIEW

File: phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java [hunk 1]
```diff
@@ -68,258 +67,250 @@
 import com.google.common.collect.ImmutableMap.Builder;
 
 public final class TypeUtil {
-	
+
     private static final Log LOG = LogFactory.getLog(TypeUtil.class);
-    private static final HBaseBinaryConverter binaryConverter = new HBaseBinaryConverter ();
```
Comment (prashantkommireddi) [inline phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java]: These are just formatting changes.

File: phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java [hunk 2]
```diff
@@ -68,258 +67,250 @@
...
+        builder.put(PUnsignedFloat.INSTANCE, DataType.FLOAT);
+        builder.put(PUnsignedInt.INSTANCE, DataType.INTEGER);
+        builder.put(PUnsignedLong.INSTANCE, DataType.LONG);
+        builder.put(PUnsignedSmallint.INSTANCE, DataType.INTEGER);
+        builder.put(PUnsignedTime.INSTANCE, DataType.DATETIME);
+        builder.put(PUnsignedTimestamp.INSTANCE, DataType.DATETIME);
+        builder.put(PUnsignedTinyint.INSTANCE, DataType.INTEGER);
         return builder.build();
     }
+
```
Comment (prashantkommireddi) [inline phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java]: Formatting changes only

File: phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java [hunk 3]
```diff
@@ -68,258 +67,250 @@
...
+                case DataType.LONG:
+                    tuple.set(i, DataType.toLong(object));
+                    break;
+                case DataType.BOOLEAN:
+                    tuple.set(i, DataType.toBoolean(object));
+                    break;
+                case DataType.DATETIME:
+                    tuple.set(i, DataType.toDateTime(object));
+                    break;
+                case DataType.BIGDECIMAL:
```
Comment (prashantkommireddi) [inline phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java]: Adding BigDecimal and BigInteger types

Comment (elilevine) [inline phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java]: Is this the only functional change in this PR? Everything else if formatting?

File: phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    public void testTransformToTuple() throws Exception {
+        PhoenixPigDBWritable record = mock(PhoenixPigDBWritable.class);
+        List<Object> values = Lists.newArrayList();
+        values.add("213123");
+        values.add(1231123);
+        values.add(31231231232131L);
+        values.add("bytearray".getBytes());
+        when(record.getValues()).thenReturn(values);
+
+        ResourceFieldSchema field = new ResourceFieldSchema().setType(DataType.CHARARRAY);
```
Comment (elilevine) [inline phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java]: Mind adding tests for other supported data types?

File: phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java (hunk 3 above)
Comment (prashantkommireddi) [inline phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java]: Yes.

File: phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java (hunk 4 above)
Comment (prashantkommireddi) [inline phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java]: I added a few for these changes as I noticed no tests existed around this whatsoever. Can we get this in, and may be open a JIRA for adding tests around other data types?

Comment (elilevine) [inline phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java]: I'm cool with the changes. @JamesRTaylor, @jyates you OK to check this in? I'm happy to commit.

File: phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java [hunk 5]
```diff
@@ -68,258 +67,250 @@
...
+    private static final HBaseBinaryConverter BINARY_CONVERTER = new HBaseBinaryConverter();
+    private static final ImmutableMap<PDataType, Byte> PHOENIX_TO_PIG_TYPE = init();
+    private static final TupleFactory TUPLE_FACTORY = TupleFactory.getInstance();
+
+    private TypeUtil() {}
+
+    /**
+     * A map of Phoenix to Pig data types.
+     * 
+     * @return
```
Comment (jyates) [inline phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java]: nit: if you are going to change this, at least move around the javadoc so it reads: "@return map of Phoenix to Pig data types."

File: phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java [hunk 6]
```diff
@@ -68,258 +67,250 @@
...
-	 * @param obj
-	 * @return PDataType
-	 */
-	public static PDataType getType(Object obj, byte type) {
-		if (obj == null) {
-			return null;
-		}
-		PDataType sqlType;
+     * This method encodes a value with Phoenix data type. It begins with checking whether an object is BINARY and makes
+     * a call to {@link #castBytes(Object, PDataType)} to convery bytes to targetPhoenixType
```
Comment (jyates) [inline phoenix-pig/src/main/java/org/apache/phoenix/pig/util/TypeUtil.java]: otherwise, it...?

**Comment by jyates** [comment]
Couple of minor comments. otherwise, lgtm

File: phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java (hunk 4 above)
Comment (JamesRTaylor) [inline phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java]: Thanks, @elilevine. I +1'ed it on the JIRA, so please commit. We're targeting a 4.4.1 next week.

Comment (elilevine) [inline phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java]: @prashantkommireddi, mind making the few small changes Jesse asked for? Would be nice to get this in on Monday. Thanks!

Comment (prashantkommireddi) [inline phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java]: @elilevine @JamesRTaylor im at a conference these 3 days. I will try making changes sometime today or tomorrow. Works?

Comment (elilevine) [inline phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java]: Works for me, thanks.

Comment (prashantkommireddi) [inline phoenix-pig/src/test/java/org/apache/phoenix/pig/util/TypeUtilTest.java]: @elilevine I have updated pull request, can you please take a look?

**Pull request closed without merging**